packages/server-wallet/src/fp.ts

```diff
@@ -0,0 +1,13 @@
+export type Left<E extends Error> = {readonly _tag: 'left'; error: E};
```

Comment:
Using https://www.npmjs.com/package/either-ts might save some of this boiler plate? Although it may be simpler to just use our own implementation

Reply:
I thought about that, but figured that implementing the boilerplate was less effort than evaluating which option we should use.

Plus, using our own implementation means that we can have more strictly typed errors (rather than just the `Error` type)

Reply:
I had a good experience using `fp-ts` in the `simple-hub` package FWIW.

Reply:
I am happy to try that!